spharpy/classes/sh.py

```diff
@@ -156,31 +257,13 @@ def __init__(
         self.condon_shortley = condon_shortley
 
     # Properties
-    @property
-    def condon_shortley(self):
-        """Get or set the Condon-Shortley phase term."""
-        return self._condon_shortley
 
-    @condon_shortley.setter
+    @SphericalHarmonicDefinition.condon_shortley.setter
     def condon_shortley(self, value):
         """Get or set the Condon-Shortley phase term."""
-        if isinstance(value, str):
-            if value != 'auto':
-                raise ValueError(
-                    "condon_shortley must be a bool or the string 'auto'")
-            # If basis_type hasn't been set yet, assume "complex" by default,
-            # but in practice __init__ sets basis_type before condon_shortley.
-            if self.basis_type == "complex":
-                resolved = True
-            else:
-                resolved = False
-            value = resolved
-        elif not isinstance(value, bool):
-            raise TypeError(
-                "condon_shortley must be a bool or the string 'auto'")
+        super(__class__, SphericalHarmonics).condon_shortley.fset(self, value)
         if value != self._condon_shortley:
             self._reset_compute_attributes()
```

Comment:
I think it must be the other way around, otherwise `value != self._condon_shortley` will always be `False`
```suggestion
        if value != self._condon_shortley:
            self._reset_compute_attributes()
            super(__class__, SphericalHarmonics).condon_shortley.fset(self, value)
```